### PR TITLE
[5.9] Add support for nullable unique indexes in SQL Server

### PR DESCRIPTION
**Pull request by jlsjonas**
see https://stackoverflow.com/a/767702/620141
Note: Supports SQL Server 2008 & up (11 years, should be safe)

Rebase on master of #27370 with extra test for multi-column index

### CODE REVIEW

File: src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php [hunk 1]
```diff
@@ -105,10 +105,11 @@ public function compilePrimary(Blueprint $blueprint, Fluent $command)
      */
     public function compileUnique(Blueprint $blueprint, Fluent $command)
     {
-        return sprintf('create unique index %s on %s (%s)',
+        return sprintf('create unique index %s on %s (%s) where "%s" is not null',
```
Comment (staudenmeir) [inline src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php]: Minor nitpick: The quotes should come from `wrap()`.

```suggestion
        return sprintf('create unique index %s on %s (%s) where %s is not null',
```

File: src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php [hunk 2]
```diff
@@ -105,10 +105,11 @@ public function compilePrimary(Blueprint $blueprint, Fluent $command)
...
             $this->wrap($command->index),
             $this->wrapTable($blueprint),
-            $this->columnize($command->columns)
+            $this->columnize($command->columns),
+            implode('" is not null and "', $command->columns)
```
Comment (staudenmeir) [inline src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php]:
```suggestion
            implode(' is not null and ', $this->wrapArray($command->columns))
```

**Comment by driesvints** [comment]
@jlsjonas please always add a detailed description to your PR comment why this is needed and what the benefits are and not just link to an external website/issue.

**Comment by jlsjonas** [comment]
👍 
The link shows that nullable unique indexes are supported since SQL Server 2008
the title pretty much says it all.

## Benefits

Being able to put unique indexes on nullable columns

example usecase: newly introduced `globaluid` on the User model (for adldap2-laravel) must be unique per user, but existing users (& non-AD/ldap users) will have the column set to null; which is perfectly valid.

Current behaviour won't allow such a change on SQL Server-based installations

The new behaviour should also match how it works by default for other db drivers and thus provide more consistency.

**Comment by driesvints** [comment]
I've reverted this PR for reasons stated here: https://github.com/laravel/framework/issues/28587

**Pull request merged**